locator/token_metadata.cc

```diff
@@ -1414,12 +1423,18 @@ future<> token_metadata_impl::calculate_pending_ranges_for_replacing(
         return make_ready_future<>();
     }
     return do_with(_replacing_endpoints, strategy.get_address_ranges(unpimplified_this),
-            [this, new_pending_ranges] (std::unordered_map<inet_address, inet_address>& replacing_endpoints,
+            [this, new_pending_ranges] (std::unordered_map<inet_address, replacing_node_status>& replacing_endpoints,
                     std::unordered_multimap<inet_address, dht::token_range>& address_ranges) {
         return do_for_each(replacing_endpoints, [this, new_pending_ranges, &address_ranges]
-                (const std::pair<gms::inet_address, gms::inet_address>& node) {
+                (const std::pair<gms::inet_address, replacing_node_status>& node) {
             auto existing_node = node.first;
-            auto replacing_node = node.second;
+            auto replacing_node = node.second.node;
+            auto is_prepare_state = node.second.is_prepare_state;
+            if (is_prepare_state) {
+                tlogger.debug("Skip update pending ranges for existing_node={}, replacing_node={}, is_prepare_state={}",
+                        existing_node, replacing_node, is_prepare_state);
+                return make_ready_future<>();
```

Comment:
Can you explain how not updating pending ranges prevents the storage proxy from contacting the node? It seems to me that if we do nothing here then the storage_proxy will contact the replacing node as a natural replica.

Reply:
I think I see it, we still mark the node as replacing, so it will be removed from natural endpoints.

Reply:
Yes.